### ISSUE TITLE
RGBToHSB returns [NaN, 0, 0] when RGB values are achromatic

### ISSUE THREAD

**Issue by pseudo-me** [30-seconds/30-seconds-of-code#1849]
improved:
const h = n === 0 ? 0 : n && v === r ? (g - b) / n : v === g ? 2 + (b - r) / n : 4 + (r - g) / n;

**Comment by ninhtran3399** [comment]
ok

**Comment by Chalarangelo** [comment]
Could you provide an example of an RGB value that doesn't work?

**Comment by pseudo-me** [comment]
My Bad. The way I opened the isssue wasn't good and I'm sorry if it sounded impolite.

For example:
`RGBToHSB(200, 200, 200)`
returns
`[NaN, 0, 78]`
because the divisor which is used to calculate the hue value is zero.

**Comment by Chalarangelo** [comment]
Yep, this seems like a valid issue and the proposed solution seems to fix it. I'll look into it a little bit more and patch it as soon as possible. Thanks for reporting it! 😄
